common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -415,6 +417,109 @@ public void onReceive(@NonNull PropertyBag propertyBag) {
+                         * The Broker controller will map the response into the broker result
+                         * and signal the future with the broker result to unblock the request.
+                         */
+
+                        Logger.verbose(
+                                methodTag,
+                                "Received result (for Account Transfer V2 request) from Broker..."
+                        );
+
+                        Telemetry.emit(
+                                new ApiStartEvent()
+                                        .putApiId(TelemetryEventStrings.Api.BROKER_COMPLETE_ACQUIRE_TOKEN_ACCOUNT_TRANSFER_V2_INTERACTIVE)
+                                        .put(TelemetryEventStrings.Key.REQUEST_CODE, propertyBag.<Integer>getOrDefault(REQUEST_CODE, -1).toString())
+                                        .put(TelemetryEventStrings.Key.RESULT_CODE, propertyBag.<Integer>getOrDefault(RESULT_CODE, -1).toString())
+                        );
+
+                        mBrokerResultFuture.setResult(PropertyBagUtil.toBundle(propertyBag));
+
+                        Telemetry.emit(
+                                new ApiEndEvent()
+                                        .putApiId(TelemetryEventStrings.Api.BROKER_COMPLETE_ACQUIRE_TOKEN_ACCOUNT_TRANSFER_V2_INTERACTIVE)
+                        );
+
+                        LocalBroadcaster.INSTANCE.unregisterCallback(RETURN_BROKER_INTERACTIVE_ACQUIRE_TOKEN_RESULT);
+                    }
+                });
+
+        // To support calling from OneAuth-MSAL, which may be initialized without an Activity
+        // add Flags to start as a NEW_TASK if we are launching from an application Context
+        brokerActivityIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        mApplicationContext.startActivity(brokerActivityIntent);
+
+        final AcquireTokenResult result;
+        try {
+            // Wait to be notified of the result being returned... we could add a timeout here if we want to
+            final Bundle resultBundle = mBrokerResultFuture.get();
+
+            final String negotiatedBrokerProtocolVersion = interactiveAccountTransferV2RequestIntent.getStringExtra(NEGOTIATED_BP_VERSION_KEY);
+
+            // TODO: Is this accurate? BrokerAccountDataManager contains MSA accounts
+            // For MSA Accounts Broker doesn't save the accounts, instead it just passes the result along,
+            // MSAL needs to save this account locally for future token calls.
+            // parameters.getOAuth2TokenCache() will be non-null only in case of MSAL native
+            // If the request is from MSALCPP , OAuth2TokenCache will be null.
+            if (parameters.getOAuth2TokenCache() != null && !BrokerProtocolVersionUtil.canSupportMsaAccountsInBroker(negotiatedBrokerProtocolVersion)) {
+                saveMsaAccountToCache(resultBundle, (MsalOAuth2TokenCache) parameters.getOAuth2TokenCache());
+            }
```

Comment:
This should not be needed anymore. This was before PRTv3 when MSA accounts where not supported in broker.

Reply:
What does "this" refer to, the whole if statement?

Reply:
saveMsaAccountToCache part